apps/gimbtonomy/LEDMatrix.h

```diff
@@ -0,0 +1,47 @@
+/* LED Matrix Header */
+
+#ifndef LED_MATRIX_H
+#define LED_MATRIX_H
+
+#include <mbed.h>
+#include <CANMsg.h>
+
+//Use whatever uint8_t red, green, blue gets assigned by the CAN messages to set the PWM of R, G, B 
+
+/*typedef union {
+	uint32_t  hexColourCode;
+		struct {
+			uint8_t green;
+			uint8_t blue;
+			uint8_t red;
+		} coloursRGB;
+} HexColours;*/
+
+class LEDMatrix {
+	
+	public:
+		//Data constructor - pins determined beforehand
+		//LEDMatrix();
+		
+		void setColour(uint8_t msgRed, uint8_t msgGreen, uint8_t msgBlue);
```

Comment:
For the LED task we had some hard coded functions to specifically show red, green and blue for the required tasks. you may want to talk to @younesr1  if he would like these functions as it may make things easier for him.

Reply:
yeah i think that'd be kinda nice to have. It's not hard to implement once you have the setColourFrequency() done. it basically just calls that function with pre-determined args.